stdlib/universe/mode.go

```diff
@@ -0,0 +1,668 @@
+	ns := new(ModeProcedureSpec)
+
+	*ns = *s
+
+	return ns
+}
+
+// TriggerSpec implements plan.TriggerAwareProcedureSpec
+func (s *ModeProcedureSpec) TriggerSpec() plan.TriggerSpec {
+	return plan.NarrowTransformationTriggerSpec{}
+}
+
+func createModeTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*ModeProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewModeTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type modeTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	column string
+}
+
+func NewModeTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *ModeProcedureSpec) *modeTransformation {
+	return &modeTransformation{
+		d:      d,
+		cache:  cache,
+		column: spec.Column,
+	}
+}
+
+func (t *modeTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *modeTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("mode found duplicate table with key: %v", tbl.Key())
+	}
+
+	colIdx := execute.ColIdx(t.column, tbl.Cols())
+	if colIdx < 0 {
+		// doesn't exist in this table, so add an empty value
+		if err := execute.AddTableKeyCols(tbl.Key(), builder); err != nil {
+			return err
+		}
+		colIdx, err := builder.AddCol(flux.ColMeta{
+			Label: execute.DefaultValueColLabel,
+			Type:  flux.TString,
+		})
+		if err != nil {
+			return err
+		}
+
+		if err := builder.AppendString(colIdx, ""); err != nil {
+			return err
+		}
+		if err := execute.AppendKeyValues(tbl.Key(), builder); err != nil {
+			return err
+		}
+		// TODO: hack required to ensure data flows downstream
+		return tbl.Do(func(flux.ColReader) error {
+			return nil
+		})
+	}
+
+	col := tbl.Cols()[colIdx]
+
+	if err := execute.AddTableKeyCols(tbl.Key(), builder); err != nil {
+		return err
+	}
+	colIdx, err := builder.AddCol(flux.ColMeta{
+		Label: execute.DefaultValueColLabel,
+		Type:  col.Type,
+	})
+	if err != nil {
+		return err
+	}
+
+	if tbl.Key().HasCol(t.column) {
+		j := execute.ColIdx(t.column, tbl.Key().Cols())
+
+		if err := builder.AppendValue(colIdx, tbl.Key().Value(j)); err != nil {
+			return err
+		}
+
+		if err := execute.AppendKeyValues(tbl.Key(), builder); err != nil {
+			return err
+		}
+		// TODO: hack required to ensure data flows downstream
+		return tbl.Do(func(flux.ColReader) error {
+			return nil
+		})
+	}
+
+	var (
+		numNil int64
+	)
+
+	switch col.Type {
+	case flux.TBool:
+		boolMode := make(map[bool]int64)
+		return tbl.Do(func(cr flux.ColReader) error {
+			l := cr.Len()
+			err := t.doBool(cr, tbl, boolMode, l, numNil, builder, colIdx)
```

Comment:
In Go, errors should typically be done like this:

```go
if err := myfunc(); err != nil {
    // handle error
}
```

Or they can be directly returned. Rarely, we need the error outside of the temporary scope created by if or there's another return parameter that we need.

In this specific scenario, you can just return directly. The if condition on line 193 doesn't change anything.